controllers/virtualmachine/virtualmachine_controller.go

```diff
@@ -77,6 +81,24 @@ func AddToManager(ctx *context.ControllerManagerContext, mgr manager.Manager) er
 			handler.EnqueueRequestsFromMapFunc(classToVMMapperFn(ctx, r.Client)))
 	}
 
+	// Filter any VMs that reference a VM Class with a spec.controllerName set
+	// to a non-empty value other than "vmoperator.vmware.com". Please note that
+	// a VM will *not* be filtered if the VM Class does not exist, does not have
+	// the field spec.controllerName, or if the field spec.controllerName is
+	// empty.
+	builder = builder.WithEventFilter(
+		kubeutil.VMForControllerPredicate(
+			r.Client,
+			// These events can be very verbose, so be careful to not log them
+			// at too high of a log level.
+			r.Logger.WithName("VMForControllerPredicate").V(8),
+			vmopv1.GroupName,
+			kubeutil.VMForControllerPredicateOptions{
+				MatchIfVMClassNotFound:            true,
+				MatchIfControllerNameFieldEmpty:   true,
+				MatchIfControllerNameFieldMissing: true,
```

Comment:
I like this pattern, it keeps the controller builder logic saner and enables unit testing of the predicate logic if need be. 💯

Reply:
This is also so the current `VirtualMachine` controller can use this predicate in _default_ mode where other VM controllers can re-use this predicate and force filter out classes that have no or an empty `spec.controllerName` field.